BranchValidator/ExtensionMethods.cs

```diff
@@ -0,0 +1,478 @@
+    /// <param name="value">The value that may or may not be located between the positions.</param>
+    /// <param name="startPos">The starting position.</param>
+    /// <param name="endPos">The ending position.</param>
+    /// <returns><c>true</c> if the <paramref name="value"/> is not in between the positions.</returns>
+    /// <remarks>
+    ///     An <paramref name="endPos"/> value that is larger than the length of this <c>string</c>
+    ///     will use the last character position for the ending position.
+    /// </remarks>
+    public static bool IsNotBetween(this string thisStr, string value, uint startPos, uint endPos)
+        => !IsBetween(thisStr, value, startPos, endPos);
+
+    /// <summary>
+    /// Returns a value indicating whether or not all occurrences of the given <c>char</c> <paramref name="value"/> are located
+    /// between the given <paramref name="leftChar"/> and <paramref name="rightChar"/>.
+    /// </summary>
+    /// <param name="thisStr">The string to process.</param>
+    /// <param name="value">The <c>char</c> value to search.</param>
+    /// <param name="leftChar">The <c>char</c> that should be to the left of the given <paramref name="value"/>.</param>
+    /// <param name="rightChar">The <c>char</c> that should be to the right of the given <paramref name="value"/>.</param>
+    /// <returns>
+    ///     True if all occurrences of the given <paramref name="value"/> are between the <paramref name="leftChar"/> and <paramref name="rightChar"/>.
+    /// </returns>
+    public static bool AllIsBetween(this string thisStr, char value, char leftChar, char rightChar)
+    {
+        if (string.IsNullOrEmpty(thisStr))
+        {
+            return false;
+        }
+
+        if (thisStr.Contains(value) is false)
+        {
+            return false;
+        }
+
+        var totalLeftChar = thisStr.Count(c => c == leftChar);
+        var totalRightChar = thisStr.Count(c => c == rightChar);
+
+        if (totalLeftChar != totalRightChar)
+        {
+            return false;
+        }
+
+        var inBounds = false;
+
+        for (var i = 0; i < thisStr.Length; i++)
+        {
+            var character = thisStr[i];
+
+            if (inBounds)
+            {
+                if (character == value)
+                {
+                    continue;
+                }
+            }
+            else
+            {
+                if (character == value)
+                {
+                    return false;
+                }
+            }
+
+            if (character == leftChar)
+            {
+                inBounds = true;
+            }
+            else if (character == rightChar)
+            {
+                inBounds = false;
+            }
+        }
+
+        return true;
+    }
+
+    /// <summary>
+    /// Returns a value indicating whether or not all of the occurrences of the given <c>char</c> <paramref name="value"/> is are not located
```

Comment:
Refactor to remove "is"
```suggestion
    /// Returns a value indicating whether or not all of the occurrences of the given <c>char</c> <paramref name="value"/> are not located
```